be/src/exec/pipeline/morsel.h

```diff
@@ -70,7 +70,7 @@ class MorselQueue {
     std::vector<MorselQueuePtr> split_by_size(size_t split_size) {
         // split_size is in (0, split_size].
         DCHECK_GT(split_size, 0);
-        DCHECK_LE(split_size, _num_morsels);
+        DCHECK(_num_morsels == 0 || split_size <= _num_morsels);
```

Comment:
can this func handle the case: _num_morsels == 0?

Reply:
@satanson  Yes, in this case,  it will the skip the loop `for (int i = 0; i < _num_morsels; ++i) {...}` and `split_morsel_queues` will contain one MorselQueue with an empty morsels.